test/convnd_bwd_data/convnd_bwd_data.cpp

```diff
@@ -0,0 +1,554 @@
+        break;
+    case 3:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv3d_bwd_data_xdl_ndhwc_kzyxc_ndhwk_f16_instances(conv_ptrs);
+        break;
+    default: break;
+    }
+}
+void GetDeviceConvBwdDataOpPtr(BF16,
+                               std::vector<DeviceConvBwdDataNoOpPtr>& conv_ptrs,
+                               int num_dim_spatial)
+{
+    switch(num_dim_spatial)
+    {
+    case 1:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv1d_bwd_data_xdl_nwc_kxc_nwk_bf16_instances(conv_ptrs);
+        break;
+    case 2:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv2d_bwd_data_xdl_nhwc_kyxc_nhwk_bf16_instances(conv_ptrs);
+        break;
+    case 3:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv3d_bwd_data_xdl_ndhwc_kzyxc_ndhwk_bf16_instances(conv_ptrs);
+        break;
+    default: break;
+    }
+}
+void GetDeviceConvBwdDataOpPtr(INT8,
+                               std::vector<DeviceConvBwdDataNoOpPtr>& conv_ptrs,
+                               int num_dim_spatial)
+{
+    switch(num_dim_spatial)
+    {
+    case 1:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv1d_bwd_data_xdl_nwc_kxc_nwk_int8_instances(conv_ptrs);
+        break;
+    case 2:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv2d_bwd_data_xdl_nhwc_kyxc_nhwk_int8_instances(conv_ptrs);
+        break;
+    case 3:
+        ck::tensor_operation::device::device_conv2d_bwd_data_instance::
+            add_device_conv3d_bwd_data_xdl_ndhwc_kzyxc_ndhwk_int8_instances(conv_ptrs);
+        break;
+    default: break;
+    }
+}
+int main(int argc, char* argv[])
+{
+    int data_type       = 1;
+    int init_method     = 1;
+    int nrepeat         = 5;
+    int num_dim_spatial = 2;
+
+    ck::conv_util::ConvParams params;
+
+    if(argc == 4)
+    {
+        data_type   = std::stoi(argv[1]);
+        init_method = std::stoi(argv[2]);
+        nrepeat     = std::stoi(argv[3]);
+    }
+    else if(argc > 4)
+    {
+        data_type       = std::stoi(argv[1]);
+        init_method     = std::stoi(argv[2]);
+        nrepeat         = std::stoi(argv[3]);
+        num_dim_spatial = std::stoi(argv[4]);
+        // check args number
+        const int preParams = 5;
+        int conv_args       = 3 + num_dim_spatial * 6;
+        int cmdline_nargs   = conv_args + preParams;
+        if(cmdline_nargs != argc)
+        {
+            PrintUseMsg();
+            return 1;
+        }
+
+        params = ParseConvParams(num_dim_spatial, argv, preParams);
+    }
+    else if(argc != 1)
+    {
+        PrintUseMsg();
+        return 1;
+    }
+
+    auto Run = [&](auto input_type, auto wei_type, auto out_type) {
```

Comment:
already defined in profile_convnd_bwd_data.cpp. so please use it.

Reply:
done